File: lib/weather/models/weather_condition.dart
enum WeatherCondition {
  clear,
  clouds,
  rain,
  snow,
  thunderstorm,
  drizzle,
  atmosphere,
  unknown,
}

extension WeatherConditionX on WeatherCondition {
  static WeatherCondition fromWeatherCode(int code) {
    if (code >= 200 && code < 300) {
      return WeatherCondition.thunderstorm;
    } else if (code >= 300 && code < 400) {
      return WeatherCondition.drizzle;
    } else if (code >= 500 && code < 600) {
      return WeatherCondition.rain;
    } else if (code >= 600 && code < 700) {
      return WeatherCondition.snow;
    } else if (code >= 700 && code < 800) {
      return WeatherCondition.atmosphere;
    } else if (code == 800) {
      return WeatherCondition.clear;
    } else if (code >= 801 && code < 900) {
      return WeatherCondition.clouds;
    } else {
      return WeatherCondition.unknown;
    }
  }

  int toWeatherCode() {
    switch (this) {
      case WeatherCondition.thunderstorm:
        return 200; // 대표 코드
      case WeatherCondition.drizzle:
        return 300;
      case WeatherCondition.rain:
        return 500;
      case WeatherCondition.snow:
        return 600;
      case WeatherCondition.atmosphere:
        return 701;
      case WeatherCondition.clear:
        return 800;
      case WeatherCondition.clouds:
        return 801;
      case WeatherCondition.unknown:
        return 0;
    }
  }

  /// 날씨 이미지 경로
  String imageAsset({bool isNight = false}) {
    if(this == WeatherCondition.clear && isNight == true) {
      return 'assets/weather_icon/moon.fill.png';
    } else if (this == WeatherCondition.clear && isNight == false) {
      return 'assets/weather_icon/clear.png';
    }

    switch (this) {
      case WeatherCondition.clear:
        return 'assets/weather_icon/clear.png';
      case WeatherCondition.clouds:
        return 'assets/weather_icon/clouds.png';
      case WeatherCondition.rain:
        return 'assets/weather_icon/rain.png';
      case WeatherCondition.snow:
        return 'assets/weather_icon/snow.png';
      case WeatherCondition.thunderstorm:
        return 'assets/weather_icon/thunderstorm.png';
      case WeatherCondition.drizzle:
        return 'assets/weather_icon/rain.png';
      case WeatherCondition.atmosphere:
        return 'assets/weather_icon/atmosphere.png';
      default:
        return 'assets/weather_icon/unknown.png';
    }
  }

  /// 날씨 텍스트
  String get label {
    switch (this) {
      case WeatherCondition.clear:
        return '맑음';
      case WeatherCondition.clouds:
        return '흐림';
      case WeatherCondition.rain:
        return '비';
      case WeatherCondition.snow:
        return '눈';
      case WeatherCondition.thunderstorm:
        return '천둥';
      case WeatherCondition.drizzle:
        return '이슬비';
      case WeatherCondition.atmosphere:
        return '안개';
      default:
        return '알 수 없음';
    }
  }
}

File: ios/Runner/Models/WeatherCondition.swift
//
//  WeatherCondition.swift
//  Runner
//
//  Created by 김두원 on 5/23/25.
//

import Foundation

enum WeatherCondition {
    case clear
    case clouds
    case rain
    case snow
    case thunderstorm
    case drizzle
    case atmosphere
    case unknown
}

extension WeatherCondition {
    static func fromWeatherCode(code: Int) -> WeatherCondition {
        switch code {
        case 200..<300: return .thunderstorm
        case 300..<400: return .drizzle
        case 500..<600: return .rain
        case 600..<700: return .snow
        case 700..<800: return .atmosphere
        case 800: return .clear
        case 801..<900: return .clouds
        default: return .unknown
        }
    }
    
    func toWeatherCode() -> Int {
        switch self {
        case .thunderstorm:
            return 200
        case .drizzle:
            return 300
        case .rain:
            return 500
        case .snow:
            return 600
        case .atmosphere:
            return 701
        case .clear:
            return 800
        case .clouds:
            return 801
        case .unknown:
            return 0
        }
    }
    
    /// 날씨 이미지 경로
    func imageAsset(isNight: Bool = false) -> String {
        if self == .clear && isNight {
            return "assets/weather_icon/moon.fill"
        } else if self == .clear && !isNight {
            return "assets/weather_icon/clear"
        }
        
        switch self {
        case .clouds:
            return "clouds"
        case .rain, .drizzle:
            return "rain"
        case .snow:
            return "snow"
        case .thunderstorm:
            return "thunderstorm"
        case .atmosphere:
            return "atmosphere"
        default:
            return "unknown"
        }
    }
    
    /// 날씨 텍스트
    var label: String {
        switch self {
        case .clear:
            return "맑음"
        case .clouds:
            return "흐림"
        case .rain:
            return "비"
        case .snow:
            return "눈"
        case .thunderstorm:
            return "천둥"
        case .drizzle:
            return "이슬비"
        case .atmosphere:
            return "안개"
        default:
            return "알 수 없음"
        }
    }
}
